src/fetch/http.cpp

```diff
@@ -422,13 +436,14 @@ struct http::impl {
   void handle_response(CURLcode res, CURL *curl);
   void free_handle(CURL *curl);
   CURL *new_handle();
-  boost::optional<fetch_error> new_request(CURL *curl, request *r);
+  boost::optional<fetch_result> new_request(CURL *curl, request *r);
   std::string url_for(unsigned int z, unsigned int x, unsigned int y) const;
   bool setup_response_tile(fetch_response &response, std::unique_ptr<std::stringstream> &stream, unsigned int z, unsigned int x, unsigned int y);
```

Comment:
Not yet, although it's possible it might need to change later. At this point the `tile` object itself only needs to know its `z, x, y` coordinates.